implementations/rust/ockam/ockam_abac/src/policy.rs

```diff
@@ -93,60 +93,53 @@ where
-            if k.find(|c: char| c.is_whitespace()).is_some() {
-                log::warn! {
-                    resource = %self.resource,
-                    action   = %self.action,
-                    id       = %id,
-                    key      = %k,
-                    "attribute key with whitespace ignored"
+        // Get identity attributes and populate the environment:
+        if let Some(attrs) = self.attributes.get_attributes(&id).await? {
+            for (key, value) in attrs.attrs() {
+                if key.find(|c: char| c.is_whitespace()).is_some() {
+                    log::warn! {
+                        resource = %self.resource,
+                        action   = %self.action,
+                        id       = %id,
+                        key      = %key,
+                        "attribute key with whitespace ignored"
+                    }
                 }
-            }
-            match str::from_utf8(v) {
-                Ok(s) => {
-                    if !self.overwrite && e.contains(k) {
-                        log::debug! {
+                match str::from_utf8(value) {
+                    Ok(s) => {
+                        if !self.overwrite && environment.contains(key) {
+                            log::debug! {
+                                resource = %self.resource,
+                                action   = %self.action,
+                                id       = %id,
+                                key      = %key,
+                                "attribute already present"
+                            }
+                            continue;
+                        }
+                        environment.put(format!("subject.{key}"), str(s.to_string()));
+                    }
+                    Err(e) => {
+                        log::warn! {
                             resource = %self.resource,
                             action   = %self.action,
                             id       = %id,
-                            key      = %k,
-                            "attribute already present"
+                            key      = %key,
+                            err      = %e,
+                            "failed to interpret attribute as string"
                         }
-                        continue;
-                    }
-                    e.put(format!("subject.{k}"), str(s.to_string()));
-                }
-                Err(e) => {
-                    log::warn! {
-                        resource = %self.resource,
-                        action   = %self.action,
-                        id       = %id,
-                        key      = %k,
-                        err      = %e,
-                        "failed to interpret attribute as string"
                     }
                 }
             }
-        }
+        };
+
+        //add the identity itself as a subject parameter
+        environment.put("subject.identity", str(id.to_string()));
```

Comment:
Should we call this `identity_id` or `identifier` for clarity?

Reply:
maybe identifier is a bit better to be more consistent